prototype/MsiWithSlc/MsiSlcPublicApis/MsiSlcPublicApis/Program.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.IdentityModel.Abstractions;
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+internal class Program
+{
+    private static async Task Main()
+    {
+        // ─── 0. Environment banner ────────────────────────────────────────────────
+        Console.WriteLine(new string('═', 70));
+        
+        // ─── 1. Certificate events ────────────────────────────────────────────────
+        ManagedIdentityApplication.BindingCertificateUpdated += c =>
+        {
+            Console.WriteLine($"\n[{Now()}] Event Fired -  BindingCertificateUpdated:");
+            PrintCertificate(c);
+        };
+
+        // ─── 2. Managed-identity source & certs ───────────────────────────────────
+        var source = await ManagedIdentityApplication.GetManagedIdentitySourceAsync()
+                                                     .ConfigureAwait(false);
+        Console.WriteLine($"\n[{Now()}] Managed Identity Source: {source}");
+
+        var cert = ManagedIdentityApplication.GetManagedIdentityBindingCertificate();
+        PrintCertificate(cert, "Initial Managed-Identity Binding Certificate");
+
+        //sleep for 5 second so we can see different timestamps
+        Thread.Sleep(5000);
+        
+        cert = ManagedIdentityApplication.ForceUpdateInMemoryCertificate();
+        PrintCertificate(cert, "Forced NEW Managed-Identity Binding Certificate");
+
+        // ─── 3. Build MI app ──────────────────────────────────────────────────────
+        IIdentityLogger logger = new IdentityLogger();
+        IManagedIdentityApplication mi = ManagedIdentityApplicationBuilder
+                                         .Create(ManagedIdentityId.SystemAssigned)
+                                         .WithExperimentalFeatures()
+                                         .WithLogging(logger, true)
+                                         .Build();
+
+        // ─── 4. Token-acquisition loop ───────────────────────────────────────────
+        string scope = "https://vault.azure.net/";
+        while (true)
+        {
+            Console.WriteLine($"\n[{Now()}] Acquiring token for scope → {scope}");
+            try
+            {
+                var result = await mi.AcquireTokenForManagedIdentity(scope)
+                                     //.WithProofOfPossession()
```

Comment:
Is this intentionally commented out?

Reply:
you can uncomment and test it when ESTS has the changes ready in PROD, as of now it is not.